### PR TITLE
[exporter/zipkin] Set "error" tag when span status is set to error

### PR DESCRIPTION
**Pull request by pmm-sumo**
**Description:** Fixes the issue with "error" tag not being set for spans that had status code set to Error. Also, makes sure that status handling for UNSET values (not filling the tags) matches the [specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk_exporters/zipkin.md#status)

**Link to tracking Issue:** #8187 

**Testing:** Unit test for the case fixed

**Documentation:** N/A

### CODE REVIEW

**Comment by pmm-sumo** [comment]
Hmm, unit test fails at:

```
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusexecreceiver	6.655s
{"level":"error","ts":1646306653.3409455,"caller":"subprocessmanager/manager.go:101","msg":"subprocess output line","output":"exit status 1","stacktrace":"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusexecreceiver/subprocessmanager.(*SubprocessConfig).pipeSubprocessOutput\n\t/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/prometheusexecreceiver/subprocessmanager/manager.go:101"}
```

File: pkg/translator/zipkin/zipkinv2/from_translator.go [hunk 1]
```diff
@@ -178,6 +170,20 @@ func spanToZipkinSpan(
 	return zs, nil
 }
 
+func populateZipkinSpanError(status pdata.SpanStatus, zs *zipkinmodel.SpanModel, tags map[string]string) {
+	tags[conventions.OtelStatusCode] = status.Code().String()
```
Comment (codeboten) [inline pkg/translator/zipkin/zipkinv2/from_translator.go:174]: Should this be checking for OK or ERROR only? https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk_exporters/zipkin.md#status

>Name of the code, either OK or ERROR. MUST NOT be set if the code is UNSET.

Comment (pmm-sumo) [reply pkg/translator/zipkin/zipkinv2/from_translator.go:174]: Good point, I will address that in the PR as well (this piece of code was essentially just moved). Also the deletion of "error":false if present

File: pkg/translator/zipkin/zipkinv2/from_translator.go [hunk 2]
```diff
@@ -178,6 +170,30 @@ func spanToZipkinSpan(
...
+	} else {
+		// The error tag should only be set if Status is Error. If a boolean version
+		// ({"error":false} or {"error":"false"}) is present, it SHOULD be removed.
+		// Zipkin will treat any span with error sent as failed.
+		delete(tags, tracetranslator.TagError)
+	}
+
+	// Per specs, Span Status MUST be reported as a key-value pair in tags to Zipkin, unless it is UNSET.
+	// In the latter case it MUST NOT be reported.
+	// https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk_exporters/zipkin.md#status
```
Comment (pmm-sumo) [inline pkg/translator/zipkin/zipkinv2/from_translator.go:185]: @codeboten this is how I read the specs here - if Status is unset, no status field (neither code nor description) should be reported. What do you think?

File: pkg/translator/zipkin/zipkinv2/from_translator.go (hunk 1 above)
Comment (pmm-sumo) [reply pkg/translator/zipkin/zipkinv2/from_translator.go:190]: Updated the code, see my other note though - I hope I read the specs right

File: pkg/translator/zipkin/zipkinv2/from_translator.go (hunk 2 above)
Comment (codeboten) [reply pkg/translator/zipkin/zipkinv2/from_translator.go:185]: That matches my expectation, thanks for fixing!

**Review by jpkrohling — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Pull request merged**